src/sat.cc

```diff
@@ -126,122 +127,98 @@ bool Sat::CheckEnvironment() {
 
   // Check if the cpu frequency test is enabled and able to run.
   if (cpu_freq_test_) {
-    if (!CpuFreqThread::CanRun()) {
-      logprintf(0,
-                "Process Error: This platform does not support this "
-                "test.\n");
-      bad_status();
+    if (!CpuFreqThread::CanRun(setup_step)) {
       return false;
     } else if (cpu_freq_threshold_ <= 0) {
-      logprintf(0,
-                "Process Error: The cpu frequency test requires "
-                "--cpu_freq_threshold set to a value > 0\n");
-      bad_status();
+      setup_step.AddError(
+          Error{.symptom = kProcessError,
+                .message = "The CPU frequency test requires "
+                           "--cpu_freq_threshold be set to a positive value."});
       return false;
     } else if (cpu_freq_round_ < 0) {
-      logprintf(0,
-                "Process Error: The --cpu_freq_round option must be greater"
-                " than or equal to zero. A value of zero means no rounding.\n");
-      bad_status();
+      setup_step.AddError(Error{
+          .symptom = kProcessError,
+          .message = "The --cpu_freq_round option must be greater than or "
+                     "equal to zero. A value of zero means no rounding."});
       return false;
     }
   }
 
   // Use all CPUs if nothing is specified.
   if (memory_threads_ == -1) {
     memory_threads_ = os_->num_cpus();
-    logprintf(7, "Log: Defaulting to %d copy threads\n", memory_threads_);
+    setup_step.AddLog(Log{
+        .severity = LogSeverity::kDebug,
+        .message = absl::StrFormat(
+            "Defaulting to using %d memory copy threads (same number as there "
+            "are CPU cores)",
+            memory_threads_)});
   }
 
   // Use all memory if no size is specified.
-  if (size_mb_ == 0) size_mb_ = os_->FindFreeMemSize() / kMegabyte;
+  if (size_mb_ == 0) size_mb_ = os_->FindFreeMemSize(setup_step) / kMegabyte;
   size_ = static_cast<int64>(size_mb_) * kMegabyte;
 
-  // Autodetect file locations.
```

Comment:
I have not context why this is removed but I will trust your judgement.

Reply:
It doesn't do anything. You can see that in the OS class the find files function doesn't have an implementation - turns out that it's only implemented in our internal google version and not the open source version. Since this is a full fork I deleted it entirely.